src/azure-cli/azure/cli/command_modules/synapse/commands.py

```diff
@@ -137,5 +169,84 @@ def get_custom_sdk(custom_module, client_factory):
         g.custom_command('list', 'list_role_definitions')
         g.custom_show_command('show', 'get_role_definition')
 
+    # Data Plane Commands --Artifacts Linked service operations
+    with self.command_group('synapse linked-service', synapse_linked_service_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('create', 'create_or_update_linked_service', supports_no_wait=True)
+        g.custom_command('set', 'create_or_update_linked_service', supports_no_wait=True)
```

Comment:
>set [](start = 26, length = 3)

Usually we use "update" which is a patch like operation. Rarely to use "set". Please refer to 
[https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#standard-command-types](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#standard-command-types) for more details.

Reply:
The things these commands do are more like "set", will replace all properties of a resource without preserving existing values. Because there have the cmdlets Set-AzSynapsexxx in powershell which alias is New-AzSynapsexxx. So I use "set" here. Maybe I can delete the "set" command directly, because it has the same function with "create". What do you think about?

Reply:
Usually CLI doesn't have 'set' command

---
In reply to: [506164081](https://github.com/Azure/azure-cli/pull/15296#discussion_r506164081) [](ancestors = 506164081)